src/main/java/com/fasterxml/jackson/databind/ser/BeanSerializerFactory.java

```diff
@@ -767,6 +769,28 @@ protected void removeSetterlessGetters(SerializationConfig config, BeanDescripti
         }
     }
 
+    /**
+     * Helper method that will remove all properties that are added by default JDK methods,
+     * e.g. java.lang.CharSequence.isEmpty() (since JDK 15).
+     */
+    private void removeDefaultProperties(List<BeanPropertyWriter> properties)
```

Comment:
I'm not quite sure about the policy with new protected methods, so I minimized the visibility for the first iteration